Fixes #930 Post header sticks to the top

cindyledev opened the pull request:
<!--
Thanks for sending a pull request!
- if this is your first time, please ensure you have read through our contributor guide: https://github.com/Seneca-CDOT/telescope/blob/master/docs/CONTRIBUTING.md
-->

## Issue This PR Addresses
Fixes #930 

<!--
1. Automatically close the issue when this PR is merged
    USAGE: Fixes #<issue number>
2. If your PR addresses an issue but does not close it
    USAGE: #<issue number> <reason>(i.e. This issue was worked on by @user and myself and his PR should close the issue.)
-->

## Type of Change

<!-- bug fix, feature, documentation, UI, etc. -->

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description

<!-- Please add a detailed description of what this PR does and why it is needed -->
Fixes the vertical spacing for post heading (I might have reduced it too much, please nit this in reviews)

@humphd has been asking for this but I didn't see an issue filed so here it is!. Post header now sticks to the top so we can see whose post it is. Very convenient for the long posts we've been getting recently.

For Desktop
![930_desktop](https://user-images.githubusercontent.com/36822198/79811051-b078ac00-8342-11ea-9b0c-b2b822764099.gif)

For Mobile
![930_mobile](https://user-images.githubusercontent.com/36822198/79811145-f6ce0b00-8342-11ea-8a28-17923d6fa1b6.gif)

Bugs introduced: 
- the scroll-to-top button now hides underneath the post headers for some reason, I've tried to set the header's zIndex to 200 so that it's lower than the button's zIndex of 300 but it didn't do anything.
- navbar also hides on scroll in other pages like the About page, not sure if we want that

Scope creep:
- changed gutter for ScrollToTop anchor from 64px to 1px

## Checklist

<!-- Before submitting a PR, address each item -->

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

humphd commented:
I *love* this idea.  So many times I've lost track of which post I was on when I leave the Telescope tab and come back.

Either in this, or a follow-up, I have some thoughts about how to improve this:

- on mobile, we end up losing a lot of reading space due to have 2 headers occupying about 1/3rd of the available space.  Could we do something to shrink this blue header on mobile?  Or just when it's in the scrolled, sticky state?  Maybe we could cut the title off a t 1 line high and use ellipses, shrink the height, etc?

- on desktop, the top-left-hand "Telescope" looks a bit odd with this.  I notice in your screencast above, you cut it off yourself.  I wonder if we can think about how to clean up the look of the header to better accommodate this new addition along the top.

Having said that, I'm happy to take this and iterate on it in follow-ups.  I'll let the frontend crew review the code, but this is amazing from a user perspective.

More fun PRs like this!

cindyledev commented:
>on mobile, we end up losing a lot of reading space due to have 2 headers occupying about 1/3rd of the available space.  Could we do something to shrink this blue header on mobile?  Or just when it's in the scrolled, sticky state?  

We can hide the navbar when we scroll down with https://material-ui.com/components/app-bar/#hide-app-bar (I believe it's suppose to reveal itself again when we scroll back up), the demo is too short to see

> Maybe we could cut the title off a t 1 line high and use ellipses, shrink the height, etc?

I thought about using ellipses for long titles too. Got the idea from your WEB422 bridge lab and your recent WEB222 traffic camera lab.

> * on desktop, the top-left-hand "Telescope" looks a bit odd with this.  I notice in your screencast above, you cut it off yourself.  I wonder if we can think about how to clean up the look of the header to better accommodate this new addition along the top.

idk maybe the top-left-hand Telescope isn't bold? I know when I modified it in #1031 , it indented the word a little bit

raygervais commented:
Responsive bug which comes about because of this: 

![image](https://user-images.githubusercontent.com/14265337/79814498-61377900-834c-11ea-8497-8667323ba2f7.png)

cindyledev commented:
Ah I might just revert my scope creep in another issue since I'm not 100% sure what else will be affected by my modification of 100vh

Also https://www.w3schools.com/howto/howto_js_shrink_header_scroll.asp this might be useful

cindyledev commented:
Pushed changes: 
- reverted vh and button position in Banner back to what it was originally
- found out that the gutter for ScrollToTop anchor was actually 64px so I reduced it to 1px
- trying hide navbar on scroll idea. It seems to do what it's suppose to scrolling down then you realize "wait how do I get the navbar back when I'm scrolled like 10 posts down? I have to scroll all the way back up to the banner to see the navbar again?" I feel like we're installing some kind of weird elevator system with all this scrolling up and down...

Need some feedback on: 
- the navbar and post header currently act independently. So I think we got the right behaviour for scrolling down but when you try to scroll back up, the navbar doesn't reveal itself as expected
- appropriate fontSizes for different screen sizes
- header height
- maybe tweak the header design a bit so Author and Date are on the same line just spaced evenly?

To do in here or in follow up:
- use ellipses for long post titles so the title will only take up one line

sorry if this doesn't make any sense... English is hard

humphd commented:
See #1039 for some other discussions about this for follow-ups.

cindyledev commented:
New changes:
- added ellipsis for long titles
- removed disableHysteresis option so scrolling up reveals the navbar now

This is getting close, just need some feedback font size, line height, styling, etc for desktop and mobile. I'm still not 100% sure what to do with the scrolling up thing because the navbar covers the post header, should we push the post header down when the navbar is visible? (idk how to do this yet)

Silvyre commented:
> Need some feedback on:
> * header height

The overall height of the sticky post headers are good, IMO.

> * maybe tweak the header design a bit so Author and Date are on the same line just spaced evenly?

Yeah, I think we can afford to space out the author and date information, e.g.
![image](https://user-images.githubusercontent.com/13500769/79923566-c601da80-8403-11ea-9766-47c9dae5409a.png)

If we do so, we can afford to shorten the header a bit more, too.

humphd commented:
@cindyledev do you want to work on the mobile version of this in a follow-up?  I think the header is way too big on mobile, but we probably shouldn't deal with that here.

In fact, I'm tempted to say that we should take this, once it passes code reviews, and iterate.  What this does is already better than what we do now.

humphd commented on src/frontend/src/components/Header/MobileHeader.jsx:
```diff
@@ -78,6 +81,26 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
+function HideOnScroll(props) {
```
This code is repeated.  Let's move it to a separate file so you can re-use the component in multiple files.

cindyledev replied:
Okay I just moved HideOnScroll to its own component. :)

cindyledev commented:
> @cindyledev do you want to work on the mobile version of this in a follow-up? I think the header is way too big on mobile, but we probably shouldn't deal with that here.

No, someone else can do it if they want. I don't know how to fix the papercuts

humphd commented on src/frontend/src/components/Header/DesktopHeader.jsx:
```diff
@@ -114,46 +115,48 @@ const Header = () => {
-        </Toolbar>
-      </AppBar>
+    <>
+      <HideOnScroll>
```
I think you should be wrapping this around https://github.com/Seneca-CDOT/telescope/blob/master/src/frontend/src/components/Header/Header.jsx#L14 vs. here and in the mobile one, so you don't have to duplicate.

cindyledev replied:
Theoretically, this is a very small and simple thing to do, it should just look like the following:
In Header.jsx
`import HideOnScroll from '../HideOnScroll';`
`...`
```
  return (
    <div>
      <HideOnScroll>
        {matches ? (
          <MobileHeader className="mobile-header" />
        ) : (
          <DesktopHeader className="header" />
        )}
      </HideOnScroll>
    </div>
  );
```

but no, I got a bunch of errors. I've been trying to fix it for like 2 hours. I got like 15 errors so I'm not gonna paste them here. I've tried React.forwardRef but have no idea if I was using it properly

humphd replied:
What are the errors?  You can paste huge text blocks in GitHub inside a details/summary block, see https://gist.github.com/ericclemmons/b146fe5da72ca1f706b2ef72a20ac39d.

cindyledev replied:
Everything builds but when I scroll down to where the Navbar is suppose to hide, the browser goes white and I get these errors

<details>
<summary>Errors</summary>

```md
:8000/commons.js:247020 404 page could not be found. Checkout https://www.gatsbyjs.org/docs/add-404-page/
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:263887 [HMR] connected
:8000/commons.js:254615 React-Hot-Loader: react-🔥-dom patch is not detected. React 16.6+ features may not work.
AppContainer @ :8000/commons.js:254615
dev.telescope.cdot.systems/user/info:1 Failed to load resource: the server responded with a status of 403 ()
:8000/commons.js:247020 Error getting user info Error
    at _callee$ (:8000/commons.js:268193)
    at tryCatch (:8000/commons.js:259967)
    at Generator.invoke [as _invoke] (:8000/commons.js:260196)
    at Generator.prototype.<computed> [as next] (:8000/commons.js:260019)
    at asyncGeneratorStep (:8000/commons.js:5805)
    at _next (:8000/commons.js:5827)
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:247020 Warning: Failed prop type: Invalid prop `children` supplied to `ForwardRef(Slide)`. Expected an element that can hold a ref. Did you accidentally use a plain function component for an element instead? For more information see https://material-ui.com/r/caveat-with-refs-guide
    in ForwardRef(Slide) (at HideOnScroll.jsx:14)
    in HideOnScroll (at Header.jsx:15)
    in div (at Header.jsx:14)
    in Header (at PageBase.js:32)
    in PageBase (created by WithStyles(PageBase))
    in WithStyles(PageBase) (at pages/index.js:156)
    in IndexPage (created by HotExportedIndexPage)
    in AppContainer (created by HotExportedIndexPage)
    in HotExportedIndexPage (created by PageRenderer)
    in PageRenderer (at query-result-store.js:86)
    in PageQueryStore (at root.js:56)
    in RouteHandler (at root.js:78)
    in div (created by FocusHandlerImpl)
    in FocusHandlerImpl (created by Context.Consumer)
    in FocusHandler (created by RouterImpl)
    in RouterImpl (created by Context.Consumer)
    in Location (created by Context.Consumer)
    in Router (created by EnsureResources)
    in ScrollContext (at root.js:69)
    in RouteUpdates (at root.js:68)
    in EnsureResources (at root.js:66)
    in LocationHandler (at root.js:124)
    in LocationProvider (created by Context.Consumer)
    in Location (at root.js:123)
    in Root (at root.js:131)
    in ThemeProvider (at TopLayout.js:18)
    in TopLayout (at gatsby-browser.js:6)
    in StaticQueryStore (at root.js:138)
    in _default (at app.js:67)
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:247020 Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of `Transition`.
    in MobileHeader (at Header.jsx:17)
    in Transition (created by ForwardRef(Slide))
    in ForwardRef(Slide) (at HideOnScroll.jsx:14)
    in HideOnScroll (at Header.jsx:15)
    in div (at Header.jsx:14)
    in Header (at PageBase.js:32)
    in PageBase (created by WithStyles(PageBase))
    in WithStyles(PageBase) (at pages/index.js:156)
    in IndexPage (created by HotExportedIndexPage)
    in AppContainer (created by HotExportedIndexPage)
    in HotExportedIndexPage (created by PageRenderer)
    in PageRenderer (at query-result-store.js:86)
    in PageQueryStore (at root.js:56)
    in RouteHandler (at root.js:78)
    in div (created by FocusHandlerImpl)
    in FocusHandlerImpl (created by Context.Consumer)
    in FocusHandler (created by RouterImpl)
    in RouterImpl (created by Context.Consumer)
    in Location (created by Context.Consumer)
    in Router (created by EnsureResources)
    in ScrollContext (at root.js:69)
    in RouteUpdates (at root.js:68)
    in EnsureResources (at root.js:66)
    in LocationHandler (at root.js:124)
    in LocationProvider (created by Context.Consumer)
    in Location (at root.js:123)
    in Root (at root.js:131)
    in ThemeProvider (at TopLayout.js:18)
    in TopLayout (at gatsby-browser.js:6)
    in StaticQueryStore (at root.js:138)
    in _default (at app.js:67)
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:27939 Uncaught TypeError: Cannot read property 'style' of null
    at Object.handleExit [as onExit] (:8000/commons.js:27939)
    at Transition.performExit (:8000/commons.js:257205)
    at Transition.performExit (:8000/commons.js:252954)
    at Transition.updateStatus (:8000/commons.js:257147)
    at Transition.updateStatus (:8000/commons.js:252954)
    at Transition.componentDidUpdate (:8000/commons.js:257108)
    at Transition.componentDidUpdate (:8000/commons.js:252944)
    at commitLifeCycles (:8000/commons.js:238350)
    at commitLayoutEffects (:8000/commons.js:241318)
    at HTMLUnknownElement.callCallback (:8000/commons.js:218703)
    at Object.invokeGuardedCallbackDev (:8000/commons.js:218752)
    at invokeGuardedCallback (:8000/commons.js:218807)
    at commitRootImpl (:8000/commons.js:241056)
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at commitRoot (:8000/commons.js:240896)
    at finishSyncRender (:8000/commons.js:240322)
    at performSyncWorkOnRoot (:8000/commons.js:240308)
    at :8000/commons.js:229604
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at flushSyncCallbackQueueImpl (:8000/commons.js:229599)
    at flushSyncCallbackQueue (:8000/commons.js:229587)
    at scheduleUpdateOnFiber (:8000/commons.js:239714)
    at dispatchAction (:8000/commons.js:234175)
    at handleScroll (:8000/commons.js:40990)
:8000/commons.js:247020 The above error occurred in the <Transition> component:
    in Transition (created by ForwardRef(Slide))
    in ForwardRef(Slide) (at HideOnScroll.jsx:14)
    in HideOnScroll (at Header.jsx:15)
    in div (at Header.jsx:14)
    in Header (at PageBase.js:32)
    in PageBase (created by WithStyles(PageBase))
    in WithStyles(PageBase) (at pages/index.js:156)
    in IndexPage (created by HotExportedIndexPage)
    in AppContainer (created by HotExportedIndexPage)
    in HotExportedIndexPage (created by PageRenderer)
    in PageRenderer (at query-result-store.js:86)
    in PageQueryStore (at root.js:56)
    in RouteHandler (at root.js:78)
    in div (created by FocusHandlerImpl)
    in FocusHandlerImpl (created by Context.Consumer)
    in FocusHandler (created by RouterImpl)
    in RouterImpl (created by Context.Consumer)
    in Location (created by Context.Consumer)
    in Router (created by EnsureResources)
    in ScrollContext (at root.js:69)
    in RouteUpdates (at root.js:68)
    in EnsureResources (at root.js:66)
    in LocationHandler (at root.js:124)
    in LocationProvider (created by Context.Consumer)
    in Location (at root.js:123)
    in Root (at root.js:131)
    in ThemeProvider (at TopLayout.js:18)
    in TopLayout (at gatsby-browser.js:6)
    in StaticQueryStore (at root.js:138)
    in _default (at app.js:67)

React will try to recreate this component tree from scratch using the error boundary you provided, AppContainer.
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:247020 TypeError: Cannot read property 'style' of null
    at Object.handleExit [as onExit] (:8000/commons.js:27939)
    at Transition.performExit (:8000/commons.js:257205)
    at Transition.performExit (:8000/commons.js:252954)
    at Transition.updateStatus (:8000/commons.js:257147)
    at Transition.updateStatus (:8000/commons.js:252954)
    at Transition.componentDidUpdate (:8000/commons.js:257108)
    at Transition.componentDidUpdate (:8000/commons.js:252944)
    at commitLifeCycles (:8000/commons.js:238350)
    at commitLayoutEffects (:8000/commons.js:241318)
    at HTMLUnknownElement.callCallback (:8000/commons.js:218703)
    at Object.invokeGuardedCallbackDev (:8000/commons.js:218752)
    at invokeGuardedCallback (:8000/commons.js:218807)
    at commitRootImpl (:8000/commons.js:241056)
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at commitRoot (:8000/commons.js:240896)
    at finishSyncRender (:8000/commons.js:240322)
    at performSyncWorkOnRoot (:8000/commons.js:240308)
    at :8000/commons.js:229604
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at flushSyncCallbackQueueImpl (:8000/commons.js:229599)
    at flushSyncCallbackQueue (:8000/commons.js:229587)
    at scheduleUpdateOnFiber (:8000/commons.js:239714)
    at dispatchAction (:8000/commons.js:234175)
    at handleScroll (:8000/commons.js:40990)
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:254646 Uncaught TypeError: Cannot read property 'style' of null
    at Object.handleExit [as onExit] (:8000/commons.js:27939)
    at Transition.performExit (:8000/commons.js:257205)
    at Transition.performExit (:8000/commons.js:252954)
    at Transition.updateStatus (:8000/commons.js:257147)
    at Transition.updateStatus (:8000/commons.js:252954)
    at Transition.componentDidUpdate (:8000/commons.js:257108)
    at Transition.componentDidUpdate (:8000/commons.js:252944)
    at commitLifeCycles (:8000/commons.js:238350)
    at commitLayoutEffects (:8000/commons.js:241318)
    at HTMLUnknownElement.callCallback (:8000/commons.js:218703)
    at Object.invokeGuardedCallbackDev (:8000/commons.js:218752)
    at invokeGuardedCallback (:8000/commons.js:218807)
    at commitRootImpl (:8000/commons.js:241056)
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at commitRoot (:8000/commons.js:240896)
    at finishSyncRender (:8000/commons.js:240322)
    at performSyncWorkOnRoot (:8000/commons.js:240308)
    at :8000/commons.js:229604
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at flushSyncCallbackQueueImpl (:8000/commons.js:229599)
    at flushSyncCallbackQueue (:8000/commons.js:229587)
    at scheduleUpdateOnFiber (:8000/commons.js:239714)
    at dispatchAction (:8000/commons.js:234175)
    at handleScroll (:8000/commons.js:40990)
:8000/commons.js:247020 The above error occurred in the <AppContainer> component:
    in AppContainer (created by HotExportedIndexPage)
    in HotExportedIndexPage (created by PageRenderer)
    in PageRenderer (at query-result-store.js:86)
    in PageQueryStore (at root.js:56)
    in RouteHandler (at root.js:78)
    in div (created by FocusHandlerImpl)
    in FocusHandlerImpl (created by Context.Consumer)
    in FocusHandler (created by RouterImpl)
    in RouterImpl (created by Context.Consumer)
    in Location (created by Context.Consumer)
    in Router (created by EnsureResources)
    in ScrollContext (at root.js:69)
    in RouteUpdates (at root.js:68)
    in EnsureResources (at root.js:66)
    in LocationHandler (at root.js:124)
    in LocationProvider (created by Context.Consumer)
    in Location (at root.js:123)
    in Root (at root.js:131)
    in ThemeProvider (at TopLayout.js:18)
    in TopLayout (at gatsby-browser.js:6)
    in StaticQueryStore (at root.js:138)
    in _default (at app.js:67)

React will try to recreate this component tree from scratch using the error boundary you provided, LocationProvider.
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:184944 Uncaught TypeError: Cannot read property 'style' of null
    at Object.handleExit [as onExit] (:8000/commons.js:27939)
    at Transition.performExit (:8000/commons.js:257205)
    at Transition.performExit (:8000/commons.js:252954)
    at Transition.updateStatus (:8000/commons.js:257147)
    at Transition.updateStatus (:8000/commons.js:252954)
    at Transition.componentDidUpdate (:8000/commons.js:257108)
    at Transition.componentDidUpdate (:8000/commons.js:252944)
    at commitLifeCycles (:8000/commons.js:238350)
    at commitLayoutEffects (:8000/commons.js:241318)
    at HTMLUnknownElement.callCallback (:8000/commons.js:218703)
    at Object.invokeGuardedCallbackDev (:8000/commons.js:218752)
    at invokeGuardedCallback (:8000/commons.js:218807)
    at commitRootImpl (:8000/commons.js:241056)
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at commitRoot (:8000/commons.js:240896)
    at finishSyncRender (:8000/commons.js:240322)
    at performSyncWorkOnRoot (:8000/commons.js:240308)
    at :8000/commons.js:229604
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at flushSyncCallbackQueueImpl (:8000/commons.js:229599)
    at flushSyncCallbackQueue (:8000/commons.js:229587)
    at scheduleUpdateOnFiber (:8000/commons.js:239714)
    at dispatchAction (:8000/commons.js:234175)
    at handleScroll (:8000/commons.js:40990)
:8000/commons.js:247020 The above error occurred in the <LocationProvider> component:
    in LocationProvider (created by Context.Consumer)
    in Location (at root.js:123)
    in Root (at root.js:131)
    in ThemeProvider (at TopLayout.js:18)
    in TopLayout (at gatsby-browser.js:6)
    in StaticQueryStore (at root.js:138)
    in _default (at app.js:67)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
__stack_frame_overlay_proxy_console__ @ :8000/commons.js:247020
:8000/commons.js:229617 Uncaught TypeError: Cannot read property 'style' of null
    at Object.handleExit [as onExit] (:8000/commons.js:27939)
    at Transition.performExit (:8000/commons.js:257205)
    at Transition.performExit (:8000/commons.js:252954)
    at Transition.updateStatus (:8000/commons.js:257147)
    at Transition.updateStatus (:8000/commons.js:252954)
    at Transition.componentDidUpdate (:8000/commons.js:257108)
    at Transition.componentDidUpdate (:8000/commons.js:252944)
    at commitLifeCycles (:8000/commons.js:238350)
    at commitLayoutEffects (:8000/commons.js:241318)
    at HTMLUnknownElement.callCallback (:8000/commons.js:218703)
    at Object.invokeGuardedCallbackDev (:8000/commons.js:218752)
    at invokeGuardedCallback (:8000/commons.js:218807)
    at commitRootImpl (:8000/commons.js:241056)
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at commitRoot (:8000/commons.js:240896)
    at finishSyncRender (:8000/commons.js:240322)
    at performSyncWorkOnRoot (:8000/commons.js:240308)
    at :8000/commons.js:229604
    at unstable_runWithPriority (:8000/commons.js:244598)
    at runWithPriority$1 (:8000/commons.js:229554)
    at flushSyncCallbackQueueImpl (:8000/commons.js:229599)
    at flushSyncCallbackQueue (:8000/commons.js:229587)
    at scheduleUpdateOnFiber (:8000/commons.js:239714)
    at dispatchAction (:8000/commons.js:234175)
    at handleScroll (:8000/commons.js:40990)
```
</details>

cindyledev replied:
I think I might have figured it out

cindyledev replied:
So `HideOnScroll.jsx` wanted:
```
  return (
    <Slide appear={false} direction="down" in={!trigger}>
      <div>{children}</div>
    </Slide>
  );
```
and didn't want
```
  return (
    <Slide appear={false} direction="down" in={!trigger}>
      {children}
    </Slide>
  );
```

The `<div>` around `{children}` is important because `<Slide>` adds a ref and props to the single child. If there are multiple children then `Slide` will be mad because `children.props` is not defined and it tries to reference `children.props.style` hence all the `style undefined` errors.

cindyledev commented:
Ugh so we lost the Slide down and up transition doing this... I'm wondering if I should just stick the HideOnScroll function inside `Header.jsx` instead of trying to make it it's own component.

humphd commented:
Should we break this problem up into separate issues/PRs?  You had it working well, but the code needed refactoring.  We could take the feature work here, and file another issue to refactor it into code that isn't repeated.  Or do you want to keep fighting with it here for a bit longer?

humphd commented:
Testing this on Zeit, we could also take this and fix the "top nav bar vanishes on scroll" bug in a follow up as well?

cindyledev commented:
Ready to reviews. Will file bugs as follow up

humphd commented:
Filed #1069, #1070 for follow-up to this so we don't have to block here.
